[zh-cn] sync scheduling-eviction/node-pressure-eviction.md (#42978)
kubernetes/website: content/zh-cn/docs/concepts/scheduling-eviction/node-pressure-eviction.md

-->
如果 Pod 是由替换失败 Pod 的{{< glossary_tooltip text="工作负载" term_id="workload" >}}资源
（例如 {{< glossary_tooltip text="StatefulSet" term_id="statefulset" >}}
或者 {{< glossary_tooltip text="Deployment" term_id="deployment" >}}）管理，
则控制平面或 `kube-controller-manager` 会创建新的 Pod 来代替被驱逐的 Pod。

<!--
### Self healing for static pods
-->
### 静态 Pod 的自我修复

<!--
If you are running a [static pod](/docs/concepts/workloads/pods/#static-pods)
on a node that is under resource pressure, the kubelet may evict that static
Pod. The kubelet then tries to create a replacement, because static Pods always
represent an intent to run a Pod on that node.
-->
如果你在面临资源压力的节点上运行静态 Pod，则 kubelet 可能会驱逐该静态 Pod。
由于静态 Pod 始终表示在该节点上运行 Pod 的意图，kubelet 会尝试创建替换该 Pod。

<!--
The kubelet takes the _priority_ of the static pod into account when creating
a replacement. If the static pod manifest specifies a low priority, and there
are higher-priority Pods defined within the cluster's control plane, and the
node is under resource pressure, the kubelet may not be able to make room for
that static pod. The kubelet continues to attempt to run all static pods even
when there is resource pressure on a node.
-->

> 创建替换时，kubelet 会考虑静态 Pod 的优先级。如果静态 Pod 清单指定了低优先级，

Reviewer: ```suggestion
创建替代 Pod 时，kubelet 会考虑静态 Pod 的优先级。如果静态 Pod 清单指定了低优先级，
```

> 并且集群的控制平面内定义了较高优先级的 Pod，并且节点面临资源压力，则 kubelet

Reviewer: ```suggestion
并且集群的控制平面内定义了优先级更高的 Pod，并且节点面临资源压力，则 kubelet
```

可能无法为该静态 Pod 腾出空间。
即使节点上存在资源压力，kubelet 也会继续尝试运行所有静态 pod。

<!--
## Eviction signals and thresholds

The kubelet uses various parameters to make eviction decisions, like the following:

- Eviction signals
- Eviction thresholds
- Monitoring intervals
-->

> ## 驱逐信号和阈值

Reviewer: ```suggestion
## 驱逐信号和阈值   {#eviction-signals-and-thresholds}
```

这一功能特性，资源不足的判定是基于 cgroup 层次结构中的用户 Pod 所处的局部及 cgroup 根节点作出的。
这个[脚本](/zh-cn/examples/admin/resource/memory-available.sh)
重现了 kubelet 为计算 `memory.available` 而执行的相同步骤。
kubelet 在其计算中排除了 inactive_file（非活动 LRU 列表上基于文件来虚拟的内存的字节数），
因为它假定在压力下内存是可回收的。

<!--
The kubelet recognizes two specific filesystem identifiers:

1. `nodefs`: The node's main filesystem, used for local disk volumes, emptyDir
   volumes not backed by memory, log storage, and more.
   For example, `nodefs` contains `/var/lib/kubelet/`.
1. `imagefs`: An optional filesystem that container runtimes use to store container
   images and container writable layers.

Kubelet auto-discovers these filesystems and ignores other node local filesystems. Kubelet
does not support other configurations.
-->
kubelet 可识别以下两个特定的文件系统标识符：

1. `nodefs`：节点的主要文件系统，用于本地磁盘卷、不受内存支持的 emptyDir 卷、日志存储等。
   例如，`nodefs` 包含 `/var/lib/kubelet/`。
1. `imagefs`：可选文件系统，供容器运行时存储容器镜像和容器可写层。

> kubelet 会自动发现这些文件系统并忽略其他节点本地文件系统。kubelet 不支持其他配置。

Reviewer: ```suggestion
kubelet 会自动发现这些文件系统并忽略节点本地的其它文件系统。kubelet 不支持其他配置。
```

### 驱逐条件 {#eviction-thresholds}

你可以为 kubelet 指定自定义驱逐条件，以便在作出驱逐决定时使用。

> 你可以设置[软](#soft-eviction-thresholds)和[硬](#hard-eviction-thresholds)驱逐阈值。

Reviewer: ```suggestion
你可以设置[软性的](#soft-eviction-thresholds)和[硬性的](#hard-eviction-thresholds)驱逐阈值。
```

为了提供自定义值，你应该分别设置所有阈值。

<!--
## Eviction monitoring interval

The kubelet evaluates eviction thresholds based on its configured `housekeeping-interval`,
which defaults to `10s`.
-->

> ## 驱逐监测间隔

Reviewer: ```suggestion
## 驱逐监测间隔   {#eviction-monitoring-interval}
```

kubelet 根据其配置的 `housekeeping-interval`（默认为 `10s`）评估驱逐条件。

<!--
## Node conditions {#node-conditions}

The kubelet reports [node conditions](/docs/concepts/architecture/nodes/#condition)
to reflect that the node is under pressure because hard or soft eviction
threshold is met, independent of configured grace periods.
-->

> ## 节点条件 {#node-conditions}

Reviewer: ```suggestion
## 节点状况 {#node-conditions}
```

kubelet 报告[节点状况](/zh-cn/docs/concepts/architecture/nodes/#condition)以反映节点处于压力之下，

> 因为满足硬或软驱逐条件，与配置的宽限期无关。

Reviewer: ```suggestion
原因是满足硬性的或软性的驱逐条件，与配置的宽限期无关。
```

| `DiskPressure`   | `nodefs.available`、`nodefs.inodesFree`、`imagefs.available` 或 `imagefs.inodesFree` | 节点的根文件系统或镜像文件系统上的可用磁盘空间和 inode 已满足驱逐条件 |
| `PIDPressure`    | `pid.available` | (Linux) 节点上的可用进程标识符已低于驱逐条件 |

> 控制平面还将这些节点条件[映射]((/zh-cn/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-nodes-by-condition))到污点。

Reviewer: ```suggestion
控制平面还将这些节点状况[映射]((/zh-cn/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-nodes-by-condition))为其污点。
```

flag, which controls how long the kubelet must wait before transitioning a node
condition to a different state. The transition period has a default value of `5m`.
-->

> ### 节点条件振荡

Reviewer: ```suggestion
### 节点状况波动   {#node-condition-oscillation}
```

在这种情况下，它会选择首先驱逐最低优先级的 Pod。

<!--
If you are running a [static pod](/docs/concepts/workloads/pods/#static-pods)
and want to avoid having it evicted under resource pressure, set the
`priority` field for that Pod directly. Static pods do not support the
`priorityClassName` field.
-->
如果你正在运行[静态 Pod](/zh-cn/docs/concepts/workloads/pods/#static-pods)
并且希望避免其在资源压力下被驱逐，请直接为该 Pod 设置 `priority` 字段。
静态 Pod 不支持 `priorityClassName` 字段。

<!--
When the kubelet evicts pods in response to inode or process ID starvation, it uses
the Pods' relative priority to determine the eviction order, because inodes and PIDs have no
requests.

The kubelet sorts pods differently based on whether the node has a dedicated
`imagefs` filesystem:
-->
当 kubelet 因 inode 或 进程 ID 不足而驱逐 Pod 时，

> 它使用 Pod 的相对优先级来确定驱逐顺序，因为 inode 和 PID 没有请求。

Reviewer: ```suggestion
它使用 Pod 的相对优先级来确定驱逐顺序，因为 inode 和 PID 没有对应的请求字段。
```

the kubelet reclaims the resource until the signal reaches the threshold of 1GiB,
and then continues to reclaim the minimum amount of 500MiB, until the available nodefs storage value reaches 1.5GiB.

Similarly, the kubelet tries to reclaim the `imagefs` resource until the `imagefs.available`
value reaches `102Gi`, representing 102 GiB of available container image storage. If the amount
of storage that the kubelet could reclaim is less than 2GiB, the kubelet doesn't reclaim anything.

The default `eviction-minimum-reclaim` is `0` for all resources.
-->
在这个例子中，如果 `nodefs.available` 信号满足驱逐条件，
kubelet 会回收资源，直到信号达到 1GiB 的条件，
然后继续回收至少 500MiB 直到信号达到 1.5GiB。

类似地，kubelet 尝试回收 `imagefs` 资源，直到 `imagefs.available` 值达到 `102Gi`，
即 102 GiB 的可用容器镜像存储。如果 kubelet 可以回收的存储量小于 2GiB，
则 kubelet 不会回收任何内容。

对于所有资源，默认的 `eviction-minimum-reclaim` 为 `0`。

<!--
## Node out of memory behavior

If the node experiences an _out of memory_ (OOM) event prior to the kubelet
being able to reclaim memory, the node depends on the [oom_killer](https://lwn.net/Articles/391222/)
to respond.

The kubelet sets an `oom_score_adj` value for each container based on the QoS for the pod.
-->

> ## 节点内存不足行为

Reviewer: ```suggestion
## 节点内存不足行为   {#node-out-of-memory-behavior}
```

这意味着低 QoS Pod 中相对于其调度请求消耗内存较多的容器，将首先被杀死。

与 Pod 驱逐不同，如果容器被 OOM 杀死，
`kubelet` 可以根据其 `restartPolicy` 重新启动它。

<!--
## Good practices {#node-pressure-eviction-good-practices}

The following sections describe good practices for eviction configuration.
-->
### 良好实践 {#node-pressure-eviction-good-practices}

> 以下部分描述了驱逐配置的良好实践。

Reviewer: ```suggestion
以下各小节阐述驱逐配置的好的做法。
```

--system-reserved=memory=1.5Gi
```

<!--
In this configuration, the `--system-reserved` flag reserves 1.5GiB of memory
for the system, which is `10% of the total memory + the eviction threshold amount`.

The node can reach the eviction threshold if a pod is using more than its request,
or if the system is using more than 1GiB of memory, which makes the `memory.available`
signal fall below 500MiB and triggers the threshold.
-->
在此配置中，`--system-reserved` 标志为系统预留了 1GiB 的内存，
即 `总内存的 10% + 驱逐条件量`。

如果 Pod 使用的内存超过其请求值或者系统使用的内存超过 `1Gi`，
则节点可以达到驱逐条件，这使得 `memory.available` 信号低于 500MiB 并触发条件。

<!--
### DaemonSets and node-pressure eviction {#daemonset}

Pod priority is a major factor in making eviction decisions. If you do not want
the kubelet to evict pods that belong to a DaemonSet, give those pods a high
enough priority by specifying a suitable `priorityClassName` in the pod spec.
You can also use a lower priority, or the default, to only allow pods from that
DaemonSet to run when there are enough resources.
-->
### DaemonSets 和节点压力驱逐  {#daemonset}

Pod 优先级是做出驱逐决定的主要因素。
如果你不希望 kubelet 驱逐属于 DaemonSet 的 Pod，
请在 Pod 规约中通过指定合适的 `priorityClassName` 为这些 Pod
提供足够高的 `priorityClass`。

> 您还可以使用较低优先级或默认优先级的 `priorityClass`，

Reviewer: ```suggestion
你还可以使用较低优先级或默认优先级，以便
```

仅在有足够资源时才运行 `DaemonSet` Pod。

<!--

> ## Known issues

Reviewer: 中文要提级，并添加 anchor。
